drivers/scmi-msg/scmi_pinctrl.c

```diff
@@ -0,0 +1,890 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (c) 2024, EPAM Systems
+ */
+#include <cdefs.h>
+#include <string.h>
+#include <errno.h>
+
+#include <drivers/pinctrl.h>
+#include <drivers/scmi-msg.h>
+#include <drivers/scmi.h>
+#include <lib/utils.h>
+#include <lib/utils_def.h>
+
+#include "common.h"
+
+#pragma weak pinctrl_get_groups_count
+#pragma weak pinctrl_get_functions_count
```

Comment:
Also, why are you defining default pinctrl function implementation inside the SCMI code? This is the wrong place.

Reply:
Actually I can rework it anyway you wish, but what exactly should i do?

Reply:
Well, this is very confusing to have pinctrl interface stubs in SCMI code. 

As far as I understand, we should not have this stubs at all: if a platform does not have pinctrl driver it should not provide pinctrl interface.

But if you want to make stubs anyways, the should live somewhere in `drivers/pinctrl/`, because everyone will except to find them here.

Reply:
fixed